src/core/3d/qgs3dviewsmanager.h

```diff
@@ -0,0 +1,101 @@
+/***************************************************************************
+    qgs3dviewsmanager.h
+    ------------------
+    Date                 : December 2021
+    Copyright            : (C) 2021 Belgacem Nedjima
+    Email                : gb underscore nedjima at esi dot dz
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGS3DVIEWSMANAGER_H
+#define QGS3DVIEWSMANAGER_H
+
+#include "qgis_core.h"
+#include "qgis_sip.h"
+#include <QObject>
+#include <QDomDocument>
+#include <QDomElement>
+#include <QMap>
+
+class QgsProject;
+
+/**
+ * \ingroup core
+ * \class Qgs3DViewsManager
+ *
+ * \brief Manages storage of a set of 3D views.
+ *
+ * QgsLayoutManager handles the storage, serializing and deserializing
+ * of 3D views. Usually this class is not constructed directly, but
+ * rather accessed through a QgsProject via QgsProject::viewsManager3D().
+ *
+ * \since QGIS 3.24
+ */
+class CORE_EXPORT Qgs3DViewsManager : public QObject
+{
+    Q_OBJECT
+
+  public:
+
+    /**
+     * Constructor for Qgs3DViewsManager. The project will become the parent object for this
+     * manager.
+     */
+    explicit Qgs3DViewsManager( QgsProject *project );
+
+    /**
+     * Reads the manager's state from a DOM element, restoring all 3D views
+     * present in the XML document
+     * \see writeXml()
+     */
+    bool readXml( const QDomElement &element, const QDomDocument &doc );
+
+    /**
+     *  Returns a DOM element representing the state of the manager.
+     *  \see readXml()
+     */
+    QDomElement writeXml( QDomDocument &doc ) const;
+
+    //! Removes and deletes all 3D views from the manager.
+    void clear();
+
+    //! Returns the DOM element representing the settings of the 3D view named \a name
+    QDomElement get3DViewSettings( const QString &name ) const;
```

Comment:
```suggestion
    /**
     * Returns the DOM element representing the settings of the 3D view named \a name.
     *
     * \note Not available in Python bindings
    QDomElement get3DViewSettings( const QString &name ) const SIP_SKIP;
```

Let's skip this one in the Python bindings -- the raw DOM element doesn't have any real value there.

Reply:
(The "get" prefix on the name here isn't QGIS standard, but I can see why it's required here. I'm fine with the exception here)